.github/workflows/docs-pr.yml

```diff
@@ -0,0 +1,222 @@
+                markdownKeywords += `### ${escapeSpecialChars(entry.name)} <a href="\#${entry.name}" id="${entry.name}"></a>\n\n`;
+
+                if (entry.name == "block-height")
+                  markdownKeywords += `{% hint style="info" %}\nThe Nakamoto hard fork will introduce a few new Clarity keywords. It's important to note that even with the new [block production mechanism](../concepts/block-production/), the \`block-height\` keyword behavior will not change. It will simply correspond to the current tenure height. This means any Clarity contracts using this keyword will be backwards compatible after the Nakamoto Upgrade.\n{% endhint %}\n\n`
+                else if (entry.name == "burn-block-height")
+                  markdownKeywords += `{% hint style="warning" %}\nThere is a bug in Clarity 3 when \`burn-block-height\` is used within an \`at-block\` expression. Normally, keywords executed within an \`at-block\` expression will return the data for that specified block. This bug causes \`burn-block-height\` to always return the burn block at the current chain tip, even within an \`at-block\` expression. This behavior affects any Clarity 3 contracts and will be fixed in a future hard fork.\n{% endhint %}\n\n`
+                else if (entry.name == "contract-caller")
+                  markdownKeywords += `{% hint style="warning" %}\nUse caution when leveraging all contract calls, particularly tx-sender and contract-caller as based on the design, you can unintentionally introduce attack surface area. [Read more](https://www.setzeus.com/community-blog-posts/clarity-carefully-tx-sender).\n{% endhint %}\n\n`
+                else if (entry.name == "tx-sender")
+                  markdownKeywords += `{% hint style="warning" %}\nUse caution when leveraging all contract calls, particularly tx-sender and contract-caller as based on the design, you can unintentionally introduce attack surface area. [Read more](https://www.setzeus.com/community-blog-posts/clarity-carefully-tx-sender).\n{% endhint %}\n\n`
+
+                markdownKeywords += `Introduced in: ${entry.min_version.replace(/(\d+)/, ' $1')}\n\n`;
+
+                if (entry.max_version !== null) {
+                  markdownKeywords += `Removed in: ${entry.max_version.replace(/(\d+)/, ' $1')}\n\n`;
+                };
+
+                markdownKeywords += `**Output: \`${entry.output_type}\`**\n\n`;
+                markdownKeywords += `**Description:**\n\n${entry.description}\n\n`;
+                markdownKeywords += "**Example:**\n\n\`\`\`\n" + `${entry.example.trim()}` + "\n\`\`\`\n\n";
+              });
+
+              writeToFile(OUTPUT_KEYWORDS_FILE_PATH, markdownKeywords);
+            };
+
+            const main = () => {
+              const jsonData = getFileContentsParsed();
+              if (jsonData) {
+                transformFunctions(jsonData);
+                transformKeywords(jsonData);
+              };
+            };
+
+            main();
```

Comment:
I disagree with his approach - a likely outcome here is that any edit in the upstream repo (i.e. to a hint or something else not handled here implicitly) would then need to be copied downstream into this workflow. In practice, this doesn't happen since any change will be reflected upon deployment, but then overwritten on stacks-core release. 

have you considered using a templating engine and/or sourcing files from the upstream repo for this step vs overwriting the content with the produced json?

I'm also not a fan of introducing custom code in a stacks-core workflow, since changing/updating it has a higher barrier to entry. the more generic these workflows are in this repo, the better.